src/oemof/solph/flows/_flow.py

```diff
@@ -250,6 +255,29 @@ def __init__(self, **kwargs):
                 + "nonconvex flows!"
             )
 
+        infinite_error_msg = (
+            "{} must be a finite value. Passing an infinite "
+            "value is not allowed."
+        )
+        if not self.investment:
+            warn_msg_no_nominal_value = (
+                "If {} is set in a flow (except InvestmentFlow), "
+                "nominal_value must be set as well.\n"
+                "Otherwise, it won't have any effect."
+            )
+            if self.nominal_value is None:
+                for attr in need_nominal_value:
+                    if kwargs.get(attr) is not None:
+                        warn(
+                            warn_msg_no_nominal_value.format(attr),
+                            debugging.SuspiciousUsageWarning,
+                        )
```

Comment:
Why is it a Warning an not an Error. The value is multiplied with the nominal value and e.g. 5 * None provokes an error. So it should be an Error.

Reply:
It should only be a problem if the `nominal_value` is still not set when the model is created. To my understanding, it is theoretically possible to leave it unset in the constructor and to set it later.

Reply:
I'd agree to what @p-snft stated, but I have no strong opinion on that. I think since in 99% of all cases, the `nominal_value` is set rightaway, throwing an Error would be fine. If you still want to update your `nominal_value` later, you are free to do so. Throwing an Error should be less prone to user errors.

Reply:
Agreed. Raising an error is the better option for most users.